3.5.1 doesn't work on first run

Version: 3.5.1

```
  dprint:
    name: Style
    runs-on: ubuntu-latest
    container: rustlang/rust:nightly-bullseye
    steps:
      - uses: actions/checkout@v3
      - uses: dprint/check@v2.2
```

```

2023-03-13T04:07:27.7209833Z ##[group]Run curl -fsSL https://dprint.dev/install.sh | sh -s  > /dev/null 2>&1
2023-03-13T04:07:27.7210270Z [36;1mcurl -fsSL https://dprint.dev/install.sh | sh -s  > /dev/null 2>&1[0m
2023-03-13T04:07:27.7210796Z [36;1mecho "/home/runner/.dprint/bin" >> $GITHUB_PATH[0m
2023-03-13T04:07:27.7215674Z shell: bash --noprofile --norc -e -o pipefail {0}
2023-03-13T04:07:27.7215948Z ##[endgroup]
2023-03-13T04:07:29.6024471Z ##[group]Run ~/.dprint/bin/dprint check
2023-03-13T04:07:29.6024791Z [36;1m~/.dprint/bin/dprint check[0m
2023-03-13T04:07:29.6025338Z shell: bash --noprofile --norc -e -o pipefail {0}
2023-03-13T04:07:29.6025620Z ##[endgroup]
2023-03-13T04:07:29.7259439Z Error creating cache directory: Error canonicalizing path /github/home/.cache/dprint/cache: No such file or directory (os error 2)
2023-03-13T04:07:29.7349984Z ##[error]Process completed with exit code 1.
2023-03-13T04:07:29.7440864Z Post job cleanup.
```

https://github.com/mythal/boluo/actions/runs/4401252074/jobs/7707367069

--- a/.github/workflows/ci.generate.ts
+++ b/.github/workflows/ci.generate.ts
@@ -168,6 +168,11 @@ const ci = {
           run: "cargo test --locked --all-features --release",
         },
         {
+          name: "Test integration",
+          if: "matrix.config.target == 'x86_64-unknown-linux-gnu'",
+          run: "./target/debug/dprint check",
+        },
+        {
           name: "Create installer (Windows x86_64)",
           uses: "joncloud/makensis-action@v2.0",
           if: "startsWith(matrix.config.os, 'windows') && startsWith(github.ref, 'refs/tags/')",

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -91,6 +91,9 @@ jobs:
       - name: Test (Release)
         if: 'matrix.config.run_tests == ''true'' && startsWith(github.ref, ''refs/tags/'')'
         run: cargo test --locked --all-features --release
+      - name: Test integration
+        if: matrix.config.target == 'x86_64-unknown-linux-gnu'
+        run: ./target/debug/dprint check
       - name: Create installer (Windows x86_64)
         uses: joncloud/makensis-action@v2.0
         if: 'startsWith(matrix.config.os, ''windows'') && startsWith(github.ref, ''refs/tags/'')'

--- a/crates/dprint/src/environment/real_environment.rs
+++ b/crates/dprint/src/environment/real_environment.rs
@@ -56,15 +56,12 @@ impl RealEnvironment {
     };
 
     // ensure the cache directory is created
-    let cache_dir = match (*CACHE_DIR).as_ref() {
+    match (*CACHE_DIR).as_ref() {
       Ok(cache_dir) => cache_dir,
       Err(err) => {
         bail!("Error creating cache directory: {:#}", err);
       }
     };
-    if let Err(err) = environment.mk_dir_all(cache_dir) {
-      bail!("Error creating cache directory: {:#}", err);
-    }
 
     Ok(environment)
   }
@@ -373,7 +370,11 @@ fn canonicalize_path(path: impl AsRef<Path>) -> Result<CanonicalizedPathBuf> {
 
 const CACHE_DIR_ENV_VAR_NAME: &str = "DPRINT_CACHE_DIR";
 
-static CACHE_DIR: Lazy<Result<CanonicalizedPathBuf>> = Lazy::new(|| canonicalize_path(get_cache_dir_internal(|var_name| std::env::var(var_name).ok())?));
+static CACHE_DIR: Lazy<Result<CanonicalizedPathBuf>> = Lazy::new(|| {
+  let cache_dir = get_cache_dir_internal(|var_name| std::env::var(var_name).ok())?;
+  std::fs::create_dir_all(&cache_dir)?;
+  canonicalize_path(cache_dir)
+});
 
 fn get_cache_dir_internal(get_env_var: impl Fn(&str) -> Option<String>) -> Result<PathBuf> {
   if let Some(dir_path) = get_env_var(CACHE_DIR_ENV_VAR_NAME) {

